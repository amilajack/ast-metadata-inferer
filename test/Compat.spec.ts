import Compat from "../src/compat";

jest.setTimeout(60000);

describe("Compat", () => {
  it("should write compat records to each api record", async () => {
    const [record, ...records] = await Compat();
    const recordsCount = records.length + 1;
    expect(recordsCount).toBeGreaterThanOrEqual(3900);
    console.log(`${recordsCount} records in compat.json`);
    expect(record).toHaveProperty("apiType");
    expect(record).toHaveProperty("type");
    expect(record).toHaveProperty("protoChain");
    expect(record).toHaveProperty("protoChainId");
    expect(record).toHaveProperty("astNodeTypes");
    expect(record).toHaveProperty("isStatic");
    expect(record).toMatchSnapshot();

    records.forEach((_record) => {
      expect(_record).toHaveProperty("apiType");
      expect(_record).toHaveProperty("type");
      expect(_record).toHaveProperty("protoChain");
      expect(_record).toHaveProperty("protoChainId");
      expect(_record).toHaveProperty("astNodeTypes");
      expect(_record).toHaveProperty("isStatic");
      expect(_record).toHaveProperty("compat");
      // Test the properties of non-deprecated APIs
      if (_record.compat.status && !_record.compat.status.deprecated) {
        expect(_record).toHaveProperty("compat.support");
        expect(_record).toHaveProperty("compat.status");
      }
    });
  });
});
