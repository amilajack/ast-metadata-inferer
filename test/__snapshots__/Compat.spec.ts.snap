// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compat should write compat records to each api record 1`] = `
Object {
  "apiType": "js-api",
  "astNodeTypes": Array [
    "NewExpression",
  ],
  "compat": Object {
    "mdn_url": "https://developer.mozilla.org/docs/Web/API/AbortController",
    "status": Object {
      "deprecated": false,
      "experimental": true,
      "standard_track": true,
    },
    "support": Object {
      "chrome": Object {
        "version_added": "66",
      },
      "chrome_android": Object {
        "version_added": "66",
      },
      "edge": Object {
        "version_added": "16",
      },
      "firefox": Object {
        "version_added": "57",
      },
      "firefox_android": Object {
        "version_added": "57",
      },
      "ie": Object {
        "version_added": false,
      },
      "opera": Object {
        "version_added": "53",
      },
      "opera_android": Object {
        "version_added": "47",
      },
      "safari": Array [
        Object {
          "version_added": "12.1",
        },
        Object {
          "notes": "Even though <code>window.AbortController</code> is defined, it doesn't really abort <code>fetch</code> requests. See <a href='https://webkit.org/b/174980'>bug 174980</a>.",
          "partial_implementation": true,
          "version_added": "11.1",
        },
      ],
      "safari_ios": Array [
        Object {
          "version_added": "12.2",
        },
        Object {
          "notes": "Even though <code>window.AbortController</code> is defined, it doesn't really abort <code>fetch</code> requests. See <a href='https://webkit.org/b/174980'>bug 174980</a>.",
          "partial_implementation": true,
          "version_added": "11.3",
        },
      ],
      "samsunginternet_android": Object {
        "version_added": "9.0",
      },
      "webview_android": Object {
        "version_added": "66",
      },
    },
  },
  "isStatic": true,
  "protoChain": Array [
    "AbortController",
  ],
  "protoChainId": "AbortController",
  "type": "js-api",
}
`;
